DataStructures/BitArray.cs

```diff
@@ -125,7 +126,7 @@ namespace DataStructures
     /// useful functions/operations to deal with this type of
     /// data structure.
     /// </summary>
-    public class BitArray : IComparable, ICloneable, IEnumerator, IEnumerable
+    public class BitArray : IComparable, ICloneable, IEnumerator<bool>, IEnumerable<bool>
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [When implementing IComparable, you should also override <, >, <=, >=.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4735456)